src/runtime_src/core/tools/xbmgmt2/SubCmdProgram.cpp

```diff
@@ -480,7 +507,7 @@ SubCmdProgram::execute(const SubCmdOptions& _options) const
 
   // -- Retrieve and parse the subcommand options -----------------------------
   std::string device = "";
```

Comment:
Not specific to this line, but this function is much too long to comprehend.

Reply:
I agree. Do you have a way to break down this method? This function is retrieving and parsing the arguments and doing some basic checks before calling the functionality (which is a different function).